rollup/sync_service.go

```diff
@@ -293,147 +293,155 @@ func (s *SyncService) Stop() error {
-			s.txLock.Unlock()
-			time.Sleep(s.pollInterval)
-			continue
-		}
-		// Compare the remote latest queue index to the local latest
-		// queue index. If the remote latest queue index is greater
-		// than the local latest queue index, be sure to ingest more
-		// enqueued transactions
-		var start uint64
-		if s.GetLatestEnqueueIndex() == nil {
-			start = 0
-		} else {
-			start = *s.GetLatestEnqueueIndex() + 1
-		}
-		end := *latest.GetMeta().QueueIndex
+		s.txLock.Unlock()
+		time.Sleep(s.pollInterval)
+	}
+}
 
-		log.Info("Polling enqueued transactions", "start", start, "end", end)
-		for i := start; i <= end; i++ {
-			enqueue, err := s.client.GetEnqueue(i)
-			if err != nil {
-				log.Error("Cannot get enqueue in loop", "index", i, "message", err)
-				continue
-			}
+func (s *SyncService) sequence() error {
+	// Only the sequencer needs to poll for enqueue transactions
+	// and then can choose when to apply them. We choose to apply
+	// transactions such that it makes for efficient batch submitting.
+	// Place as many L1ToL2 transactions in the same context as possible
+	// by executing them one after another.
+	latest, err := s.client.GetLatestEnqueue()
+	if err != nil {
+		return err
+	}
 
-			if enqueue == nil {
-				log.Debug("No enqueue transaction found")
-				break
-			}
+	// This should never happen unless the backend is empty
+	if latest == nil {
+		return errors.New("No enqueue transactions found")
+	}
 
-			// This should never happen
-			if enqueue.L1BlockNumber() == nil {
-				log.Error("No blocknumber for enqueue", "index", i, "timestamp", enqueue.L1Timestamp(), "blocknumber", enqueue.L1BlockNumber())
-				continue
-			}
+	// Compare the remote latest queue index to the local latest
+	// queue index. If the remote latest queue index is greater
+	// than the local latest queue index, be sure to ingest more
+	// enqueued transactions
+	var start uint64
+	if s.GetLatestEnqueueIndex() == nil {
+		start = 0
+	} else {
+		start = *s.GetLatestEnqueueIndex() + 1
+	}
+	end := *latest.GetMeta().QueueIndex
 
-			// Update the timestamp and blocknumber based on the enqueued
-			// transactions
-			if enqueue.L1Timestamp() > s.GetLatestL1Timestamp() {
-				ts := enqueue.L1Timestamp()
-				bn := enqueue.L1BlockNumber().Uint64()
-				s.SetLatestL1Timestamp(ts)
-				s.SetLatestL1BlockNumber(bn)
-				log.Info("Updated Eth Context from enqueue", "index", i, "timestamp", ts, "blocknumber", bn)
-			}
+	log.Info("Polling enqueued transactions", "start", start, "end", end)
+	for i := start; i <= end; i++ {
+		enqueue, err := s.client.GetEnqueue(i)
+		if err != nil {
+			log.Error("Cannot get enqueue in loop", "index", i, "message", err)
+			continue
+		}
 
-			log.Debug("Applying enqueue transaction", "index", i)
-			err = s.applyTransaction(enqueue)
-			if err != nil {
-				log.Error("Cannot apply transaction", "msg", err)
-			}
+		if enqueue == nil {
+			log.Debug("No enqueue transaction found")
+			break
```

Comment:
This `break` is required so that the timestamp can be updated still, its too many lines of code away from that logic so I think this: https://github.com/ethereum-optimism/go-ethereum/pull/289/files#diff-c5899bff9c70fa510017fbbb32627b6cac7e41e35eea9b392a3bdd8dd3824afbR432-R442 is perfect to be broken into another function called after `sequence`. I don't want to scope creep this PR but it would be nice to have

Reply:
https://github.com/ethereum-optimism/go-ethereum/pull/289/commits/8e6dd96af24255787d14f2378ae05631567a12c1